python/lsst/dax/apdb/scripts/metadata.py

```diff
@@ -97,3 +97,27 @@ def metadata_show(config: str, use_json: bool) -> None:
     else:
         for key, value in apdb.metadata.items():
             print(f"{key}: {value}")
+
+
+def check_instrument(instrument: str) -> None:
+    """Check that a fully-qualified instrument name is valid, if pipe_base is
+    available.
+
+    Parameters
+    ----------
+    instrument : str
+        Name of instrument to check for validity.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if the instrument is not known to `~lsst.pipe.base.Instrument`.
+    """
+    try:
+        import lsst.pipe.base
+
+        lsst.pipe.base.Instrument.from_string(instrument)
```

Comment:
I wonder if it would be a lesser evil to just do the dynamic import directly (`lsst.utils.doImportType`). That way you'd cut out the entirely artificial dependency on `pipe_base`. But you'd still need `pipe_base` if you want to check the inheritance relation, so maybe not worth it.